acis_thermal_check/state_builder.py

```diff
@@ -287,105 +280,19 @@ def get_prediction_states(self, tbegin):
-                # Now point the operative ofls directory to the Continuity directory
-                present_ofls_dir = cont_load_path
-
-            #---------------------- STOP ----------------------------------------
-            # If the load type is "STOP" then grab the continuity command
-            # set and concatenate those commands to the start of bs_cmds
-            # Take into account the SCS-107 commands which shut ACIS down
-            # and any LTCTI run
-            elif present_load_type.upper() == 'STOP':
-
-                # Obtain the continuity load commands
-                cont_bs_cmds, cont_bs_name = self.BSC.get_bs_cmds(cont_load_path)
-
-                # CombineSTOP the continuity commands with the bs_cmds
-                bs_cmds = self.BSC.CombineSTOP(cont_bs_cmds, bs_cmds, scs107_date )
-
-                # Reset the backstop collection start time for the While loop
-                bs_start_time = bs_cmds[0]['time']
-                # Now point the operative ofls directory to the Continuity directory
-                present_ofls_dir = cont_load_path
-
-            #---------------------- SCS-107 ----------------------------------------
-            # If the load type is "STOP" then grab the continuity command
-            # set and concatenate those commands to the start of bs_cmds
-            # Take into account the SCS-107 commands which shut ACIS down
-            # and any LTCTI run
-            elif present_load_type.upper() == 'SCS-107':
-                # Obtain the continuity load commands
-                cont_bs_cmds, cont_bs_name = self.BSC.get_bs_cmds(cont_load_path)
-                # Store the continuity bs commands as a chunk in the chunk list
-
-                # Obtain the CONTINUITY load Vehicle-Only file
-                vo_bs_cmds, vo_bs_name = self.BSC.get_vehicle_only_bs_cmds(cont_load_path)
-
-                # Combine107 the continuity commands with the bs_cmds
-                bs_cmds = self.BSC.Combine107(cont_bs_cmds, vo_bs_cmds, bs_cmds, scs107_date)
-
-                # Reset the backstop collection start time for the While loop
-                bs_start_time = bs_cmds[0]['time']
-                # Now point the operative ofls directory to the Continuity directory
-                present_ofls_dir = cont_load_path
-
-        # Convert backstop commands from a list of dict to a CommandTable and
-        # store in self.
-        bs_cmds = kadi.commands.CommandTable(bs_cmds)
+        # Ask Backstop History to assemble the history for this loa
+        self.BSC.Assemble_History(self.backstop_file, tbegin, self.interrupt)
+
+        # Read in the assembled history file as kadi commands
+        bs_cmds =  kadi.commands.get_cmds_from_backstop( self.BSC.assembled_hist_file_path)
+
+        bs_cmds['time'] = CxoTime(bs_cmds['date']).secs
+
         self.bs_cmds = bs_cmds
 
-        # Clip commands to tbegin
-        bs_cmds = bs_cmds[bs_cmds['date'] > tbegin]
```

Comment:
So in the old version we were clipping commands to `tbegin`. Based on your changes, do we not want to do this anymore, or is in being done in `backstop_history` now?

Reply:
Ok--sounds good. Probably a good idea in the future to revisit the -5 element backoff and find something to do that maybe makes more sense (or at least is easier to understand)